erts/emulator/sys/common/erl_check_io.c

```diff
@@ -2955,7 +2955,7 @@ erts_check_io_debug(ErtsCheckIoDebugInfo *ciodip)
     erts_dsprintf(dsbufp, "internal fds=%d\n", counters.internal_fds);
 #endif
     erts_dsprintf(dsbufp, "---------------------------------------------------------\n");
-    erts_send_error_to_logger_nogl(dsbufp);
+    /* erts_send_error_to_logger_nogl(dsbufp); */
```

Comment:
I don't think we want to remove this error logger report completely as it helps a lot when trying to debug very rare race conditions in these testcases.

I wonder if we could use the `cth_log_redirect` on test suites that emit a lot of these warnings? That way they should end up in those testcase logs instead of stdout.

Reply:
Maybe instead of printing this information it could be returned? So whenever test case fails, it gets printed, but otherwise it does not. My primary problem is that when `ts:run(emulator, [batch])` is used for testing, it generates >20Mb log files that aren't working well with things like GitLab.

Reply:
Yes, doing it like that would also be an acceptable solution.

What is the limit that gitlab has? You cannot have too large files, or is it the console log that cannot be too big?

Reply:
Hard limit is 20Mb, with 500kb displayed. Files can be of any size, but console output is limited.
I found that printing this debug info can only be useful when test case is about to fail, hence I changed the diff to send the error log only when error is actually detected.

With cth_surefire support this change may not be relevant, but I still think it'd be good to have it for manual test runs.

Reply:
Works for me too, output still fits in 500kb displayed.